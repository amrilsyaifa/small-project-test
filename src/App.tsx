import React from 'react';
import AppRouter from './App.Router';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import { persistor, store } from 'src/Store';

const App = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <AppRouter />
      </PersistGate>
    </Provider>
  );
};

export default App;
