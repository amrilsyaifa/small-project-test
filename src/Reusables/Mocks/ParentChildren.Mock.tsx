import React from 'react';

/**
 * Separate children from Component so it's not circular referencing when stringified
 */
interface IParentChildrenSeparator {
  mockLabel: string;
  children: React.ReactNode;
  MockElement?: React.FC;
}
export const ParentChildrenSeparator: React.FC<IParentChildrenSeparator> = ({
  mockLabel,
  children,
  MockElement,
  ...props
}) => {
  const RenderElement = MockElement ? MockElement : MockDivElement;

  return (
    <RenderElement data-testid={mockLabel} data-props={JSON.stringify(props)}>
      {children}
    </RenderElement>
  );
};

export const MockDivElement = ({ ...props }) => <div {...props} />;

export default ParentChildrenSeparator;
