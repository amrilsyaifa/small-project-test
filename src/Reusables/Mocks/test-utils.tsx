import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { persistedReducer } from 'src/Store';
// Import your own reducer

interface IWrapper {
  children: React.ReactNode;
}

const render = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  { store = configureStore({ reducer: persistedReducer }), ...renderOptions } = {}
) => {
  const Wrapper: React.FC<IWrapper> = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
