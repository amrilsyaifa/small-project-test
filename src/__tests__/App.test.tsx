import { render, cleanup } from '@testing-library/react';
import React from 'react';
import { MatchMediaObject } from 'src/Reusables/Mocks/MatchMediaObjectTest';
import App from '../App';

afterEach(cleanup);
beforeEach(() => {
  MatchMediaObject();
});

describe('App', () => {
  it('should render expect snapshoot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
