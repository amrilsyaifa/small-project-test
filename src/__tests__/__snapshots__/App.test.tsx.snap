// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render expect snapshoot 1`] = `
<div>
  <div
    class="layout-content"
  >
    <section
      class="ant-layout"
    >
      <div
        class="container"
      >
        <main
          class="ant-layout-content"
          style="padding: 0px 50px;"
        >
          <div
            class="site-layout-content"
          >
            <div>
              <li
                class="ant-list-item"
              >
                <div
                  class="container"
                >
                  <div
                    class="ant-card ant-card-bordered card"
                    style="width: 300px;"
                  >
                    <div
                      class="ant-card-head"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          <div>
                            sunt aut facere repellat provident occaecati excepturi optio reprehenderit
                          </div>
                        </div>
                        <div
                          class="ant-card-extra"
                        >
                          <div
                            class="wrapper-header-action"
                          >
                            <div
                              class="edit-icon"
                              data-testid="edit-btn"
                            >
                              <span
                                aria-label="edit"
                                class="anticon anticon-edit"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="edit"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="remove-icon"
                              data-testid="remove-btn"
                            >
                              <span
                                aria-label="delete"
                                class="anticon anticon-delete"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="delete"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    >
                      <div>
                        quia et suscipit\\nsuscipit recusandae consequuntur expedita et cum\\nreprehenderit molestiae ut ut quas totam\\nnostrum rerum est autem sunt rem eveniet architecto
                      </div>
                    </div>
                  </div>
                </div>
              </li>
            </div>
          </div>
        </main>
      </div>
    </section>
  </div>
</div>
`;
